notebooks/model_conversion-classification.ipynb

```diff
@@ -0,0 +1,656 @@
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "X, y = datasets.load_breast_cancer(return_X_y=True)\n",
+    "X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, y, \n",
+    "                                      test_size=0.3, random_state=random_state)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1518b910",
+   "metadata": {
+    "papermill": {
+     "duration": 0.008037,
+     "end_time": "2023-04-03T11:44:07.695713",
+     "exception": false,
+     "start_time": "2023-04-03T11:44:07.687676",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# Initializing and fitting the models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "b027f575",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-04-03T11:44:07.714843Z",
+     "iopub.status.busy": "2023-04-03T11:44:07.713799Z",
+     "iopub.status.idle": "2023-04-03T11:44:07.719765Z",
+     "shell.execute_reply": "2023-04-03T11:44:07.718876Z"
+    },
+    "papermill": {
+     "duration": 0.017737,
+     "end_time": "2023-04-03T11:44:07.722001",
+     "exception": false,
+     "start_time": "2023-04-03T11:44:07.704264",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "RF = RandomForestClassifier(random_state = random_state, max_depth=10)\n",
+    "SVC = LinearSVC(random_state = random_state)\n",
+    "KNN = KNeighborsClassifier()\n",
+    "sklearn_models = [RF, SVC, KNN]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "906dafe3",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-04-03T11:44:07.741970Z",
+     "iopub.status.busy": "2023-04-03T11:44:07.740800Z",
+     "iopub.status.idle": "2023-04-03T11:44:07.957647Z",
+     "shell.execute_reply": "2023-04-03T11:44:07.956199Z"
+    },
+    "papermill": {
+     "duration": 0.230968,
+     "end_time": "2023-04-03T11:44:07.961665",
+     "exception": false,
+     "start_time": "2023-04-03T11:44:07.730697",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/conda/lib/python3.7/site-packages/sklearn/svm/_base.py:1208: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.\n",
+      "  ConvergenceWarning,\n"
+     ]
+    }
+   ],
+   "source": [
+    "for model in sklearn_models:\n",
+    "    model.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "36dca6c9",
+   "metadata": {
+    "papermill": {
+     "duration": 0.008582,
+     "end_time": "2023-04-03T11:44:07.980088",
+     "exception": false,
+     "start_time": "2023-04-03T11:44:07.971506",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# Convert the models to torchscript using Hummingbird and test the results (visualized further below)"
```

Comment:
why are these results visualized if they're just scalar values? can't we just check that the metrics are equal, as in the notebook below?

Reply:
You're right. But since we have 3 algorithms, 3 different models for each of them, and 3 metrics for each one too (accuracy, precision, recall), visualizing the results seemed like the more convenient choice. 

In PCA notebooks, since we don't have metrics to measure PCA results, I just compared pairwise elements of the matrices.

In this case, it would require too many calls to isclose(), 27 to be exact.

Reply:
We can reduce it to 9 if we make an array of metrics for each algorithm.

Reply:
I disagree. Plots are not the right way to compare 3 scalar values, even if that comparison happens multiple times. The overall number of calls to `allclose()` here is really not that much. You could write a function that runs comparisons given a list of models and metrics, returning `True` if all are close and `False` otherwise, and apply that. Alternatively, you could summarize the results in a `pd.DataFrame`.